Makefile

```diff
@@ -16,23 +16,42 @@ help:
 
 install:
 	@echo "... setting up virtualenv"
-	python3 -m venv sphinxenv
-	. $(VENV); pip install --upgrade -r requirements.txt
+	python3 -m venv .sphinx/venv
+	. $(VENV); pip install --upgrade -r .sphinx/requirements.txt
 
 	@echo "\n" \
 		"--------------------------------------------------------------- \n" \
 		"* watch, build and serve the documentation: make run \n" \
-		"* check spelling: make spelling \n" \
+                "* only build: make html \n" \
+                "* only serve: make serve \n" \
+                "* clean built doc files: make clean-doc \n" \
+                "* clean full environment: make clean \n" \
+		"* check spelling: make spellcheck \n" \
 		"\n" \
 		"enchant must be installed in order for pyenchant (and therefore \n" \
 		"spelling checks) to work. \n" \
 		"--------------------------------------------------------------- \n"
 run:
-	. $(VENV); sphinx-autobuild "$(SOURCEDIR)" "$(BUILDDIR)"
+	. $(VENV); sphinx-autobuild -c . "$(SOURCEDIR)" "$(BUILDDIR)"
+
+html:
+	. $(VENV); $(SPHINXBUILD) -c . "$(SOURCEDIR)" "$(BUILDDIR)" -w .sphinx/warnings.txt
+
+serve:
+	cd "$(BUILDDIR)"; python3 -m http.server 8000
+
+clean: clean-doc
+	rm -rf .sphinx/venv
+
+clean-doc:
+	rm -rf "$(BUILDDIR)"
```

Comment:
This turned into a very effective footgun for me once, when I managed to lose the variable in the Makefile and found myself running `rm -rf` on my entire hard disk. Is there a way it could be made safer?

Reply:
The safest way would be to get rid of the `$(BUILDDIR)` variable and hardcode the output directory, but I'm not sure if we want to do that?
I updated it to a solution that uses Git now - this will only delete the (untracked) files if we're in a Git repo. So it would at least not delete a full disk. ;)